Streetcode/Streetcode.XUnitTest/MediatRTests/Teams/GetByIdTeamTest.cs

```diff
@@ -0,0 +1,141 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.Team;
+using Streetcode.BLL.MediatR.Team.GetById;
+using Streetcode.DAL.Entities.Team;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Streetcode.XUnitTest.MediatRTests.Teams;
+
+public class GetTeamByIdTest
+{
+    private Mock<IRepositoryWrapper> _mockRepository;
+    private Mock<IMapper> _mockMapper;
+
+    public GetTeamByIdTest()
+    {
+        _mockMapper = new Mock<IMapper>();
+        _mockRepository = new Mock<IRepositoryWrapper>();
+    }
+
+    [Fact]
+    public async Task ShouldReturnSuccessfully_ExistingId()
+    {
+        //Arrange
+        var testTeam = GetTeam();
+
+        _mockRepository.Setup(x => x.TeamRepository
+            .GetSingleOrDefaultAsync(
+               It.IsAny<Expression<Func<TeamMember, bool>>>(),
+                It.IsAny<Func<IQueryable<TeamMember>,
+                IIncludableQueryable<TeamMember, object>>>()))
+            .ReturnsAsync(testTeam);
+
+        _mockMapper
+            .Setup(x => x
+            .Map<TeamMemberDTO>(It.IsAny<TeamMember>()))
+            .Returns(GetTeamDTO());
+
+        var handler = new GetByIdTeamHandler(_mockRepository.Object, _mockMapper.Object);
+
+        //Act
+        var result = await handler.Handle(new GetByIdTeamQuery(testTeam.Id), CancellationToken.None);
+
+        //Assert
+        Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.True(result.IsSuccess),
+            () => Assert.Equal(result.Value.Id, testTeam.Id)
+        );
+    }
+
+    [Fact]
+    public async Task ShouldReturnErrorResponse_NotExistingId()
+    {
+        //Arrange
+        var testTeam = GetTeam();
+        var expectedError = $"Cannot find any team with corresponding id: {testTeam.Id}";
+
+        _mockRepository.Setup(x => x.TeamRepository
+            .GetSingleOrDefaultAsync(
+               It.IsAny<Expression<Func<TeamMember, bool>>>(),
+                It.IsAny<Func<IQueryable<TeamMember>,
+                IIncludableQueryable<TeamMember, object>>>()))
+            .ReturnsAsync(GetTeamWithNotExistingId());
```

Comment:
you should make a separate method for this lines.

Reply:
if a lambda expression is a large, I should always break it into a method?

Reply:
it's not about lambda, it's about the setup function.